datasetinsights/estimators/faster_rcnn.py

```diff
@@ -419,12 +435,15 @@ def evaluate_per_epoch(
             loss_dict_reduced = reduce_dict(loss_dict)
             losses = sum(loss for loss in loss_dict_reduced.values())
             loss_metric.update(avg_loss=losses.item(), batch_size=len(targets))
+            # refresh mlflow token
+            MLFlowTracker.refresh_token()
```

Comment:
It's not clear why this has to be called in faster_rcnn.py. Can this be handled by `MLFlowTracker` class itself?

Reply:
Token expires after 4 or 5 epochs, and throws expedition:
mlflow.exceptions.MlflowException: API request to endpoint /api/2.0/mlflow/runs/log-metric failed with error code 401 != 200. Response body: 'OpenID Connect token expired: JWT has expired'

-  So we need to refresh the token, evaluate_per_epoch is common in train and evaluate so refreshing token at this place.

Reply:
Can the token refresh handled by MLFlowTracker class itself? Why do user need to remember refreshing tokens for MLFlow tracker?

Reply:
as we discussed, let's leave the responsibility of refreshing the token entirely to the tracker object and not the client (i.e. estimator in this case)